glean_parser/lint.py

```diff
@@ -286,6 +286,23 @@ def check_old_event_api(
         yield ("The old event API is gone. Extra keys require a type.")
 
 
+def check_metric_on_events_lifetime(
+    metric: metrics.Metric, parser_config: Dict[str, Any]
+) -> LintGenerator:
+    """A non-event metric on the Events ping only makes sense if its value
+    is immutable over the life of the ping."""
+    if (
+        "events" in metric.send_in_pings
+        and "all_pings" not in metric.send_in_pings
+        and metric.type != metrics.Event
```

Comment:
`metric.type` is a string, we can't compare that with a class.
```suggestion
        and metric.type != "event"
```

or (I think?)

```suggestion
        and not isinstance(metric, metrics.Event)
```

the former is more stringy, but IMO reads better